plugins/restplugin/rest_handlers.go

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restplugin
+
+import (
+	"encoding/json"
+	"git.fd.io/govpp.git/core/bin_api/vpe"
+	aclplugin "github.com/ligato/vpp-agent/plugins/defaultplugins/aclplugin/vppcalls"
+	ifplugin "github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/vppdump"
+	l2plugin "github.com/ligato/vpp-agent/plugins/defaultplugins/l2plugin/vppdump"
+	"github.com/unrolled/render"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+)
+
+//interfacesGetHandler - used to get list of all interfaces
+func (plugin *RESTAPIPlugin) interfacesGetHandler(formatter *render.Render) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+
+		plugin.Deps.Log.Info("Getting list of all interfaces")
+
+		// create an API channel
+		ch, err := plugin.Deps.GoVppmux.NewAPIChannel()
+		if err != nil {
+			plugin.Deps.Log.Errorf("Error: %v", err)
+			formatter.JSON(w, http.StatusInternalServerError, nil)
+		} else {
+			res, err := ifplugin.DumpInterfaces(plugin.Deps.Log, ch, nil)
+			if err != nil {
+				plugin.Deps.Log.Errorf("Error: %v", err)
+				formatter.JSON(w, http.StatusInternalServerError, nil)
+			} else {
+				plugin.Deps.Log.Debug(res)
+				formatter.JSON(w, http.StatusOK, res)
+			}
+		}
+		defer ch.Close()
```

Comment:
you should call defer right after the channel is created (on line 37 in this case). Otherwise, if something goes wrong in the if/else statement (nil pointer, panic, etc), defer won't be called and the channel remains open

check in every method

Reply:
done